gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala

```diff
@@ -75,7 +76,25 @@ trait BasicScanExecTransformer extends TransformSupport {
     )
   }
 
+  def unsupportedDataType () : Boolean = {
+    schema.fields.map(_.dataType).collect{
+      case byte: ByteType =>
+      case array: ArrayType =>
+      case bool: BooleanType =>
+    }.nonEmpty
+  }
+
   override def doValidate(): Boolean = {
+    // TODO need also check orc file format
+    ConverterUtils.getFileFormat(this) match {
+      case ParquetReadFormat =>
+        if (BackendsApiManager.getBackendName.equals("velox") &&
+          unsupportedDataType()) {
```

Comment:
Could you explain why this part of logic is not added into native validator?

Reply:
And maybe `GlutenConfig.getConf.isVeloxBackend` can work here.

Reply:
It is more easier to do get the data type in java side than native side. 
`GlutenConfig.getConf.isVeloxBackend` is removed in latest code.

Reply:
It looks good to me as a temporary solution, but in the long term, native validator should still be responsible to tell Gluten whether the backend supports the computing.

Reply:
Ok. I will add a TODO here.